src/macaron/slsa_analyzer/analyze_context.py

```diff
@@ -52,6 +52,8 @@ class ChecksOutputs(TypedDict):
     """The commit digest extracted from provenance, if applicable."""
     provenance_verified: bool
     """True if the provenance exists and has been verified against a signed companion provenance."""
+    provenance_l3_verified: bool
+    """True if the provenance exists and has been verified to the SLSA level 3 standard."""
```

Comment:
I think the provenance-related metadata we are storing in the context object could use some abstraction. We can use (with adjustments if necessary) the `[Provenance` mappings](https://github.com/oracle/macaron/blob/5c6c9138fbbc01c19faaf8bcadf0e5cc09ba1149/src/macaron/database/table_definitions.py#L467-L607) in our data model. We can also associate a level for each provenance, and the `provenance_verified` would then indicate if the provenance has been verified together with the evidence.

https://github.com/oracle/macaron/blob/5c6c9138fbbc01c19faaf8bcadf0e5cc09ba1149/src/macaron/slsa_analyzer/analyze_context.py#L47-L54

We should also add back and store info about the assets: https://github.com/oracle/macaron/issues/819